cdap-tms/src/test/java/co/cask/cdap/messaging/store/MessageTableTest.java

```diff
@@ -52,18 +52,36 @@ public void testSingleMessage() throws Exception {
       table.store(entryList.iterator());
       byte[] messageId = new byte[MessageId.RAW_ID_SIZE];
       MessageId.putRawId(0L, (short) 0, 0L, (short) 0, messageId, 0);
-      CloseableIterator<MessageTable.Entry> iterator = table.fetch(topicId, new MessageId(messageId), true, 50, null);
-      Assert.assertTrue(iterator.hasNext());
-      MessageTable.Entry entry = iterator.next();
-      Assert.assertArrayEquals(Bytes.toBytes(payload), entry.getPayload());
-      Assert.assertFalse(iterator.hasNext());
-      iterator = table.fetch(topicId, 0, 50, null);
-      entry = iterator.next();
-      Assert.assertArrayEquals(Bytes.toBytes(payload), entry.getPayload());
-      Assert.assertFalse(iterator.hasNext());
+
+      try (CloseableIterator<MessageTable.Entry> iterator = table.fetch(topicId,
+                                                                        new MessageId(messageId), false, 50, null)) {
+        // Fetch not including the first message, expect empty
+        Assert.assertFalse(iterator.hasNext());
```

Comment:
This is what I wrote in the javadoc:

```java
   * @param inclusive indicate whether to include the given {@link MessageId} as the first message (if still available) or not.
```java

I think the current implementation has a bug that. Fixed that in the latest commit.